client/state/data-layer/wpcom/concierge/schedules/appointments/index.js

```diff
@@ -8,129 +8,56 @@ import { translate } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
+import { mergeHandlers } from 'state/action-watchers/utils';
 import { http } from 'state/data-layer/wpcom-http/actions';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
-import { updateConciergeBookingStatus } from 'state/concierge/actions';
 import { errorNotice } from 'state/notices/actions';
-import {
-	CONCIERGE_APPOINTMENT_CANCEL,
-	CONCIERGE_APPOINTMENT_CREATE,
-	CONCIERGE_APPOINTMENT_RESCHEDULE,
-} from 'state/action-types';
-import {
-	CONCIERGE_STATUS_BOOKED,
-	CONCIERGE_STATUS_BOOKING,
-	CONCIERGE_STATUS_BOOKING_ERROR,
-	CONCIERGE_STATUS_CANCELLED,
-	CONCIERGE_STATUS_CANCELLING,
-	CONCIERGE_STATUS_CANCELLING_ERROR,
-	CONCIERGE_ERROR_NO_AVAILABLE_STAFF,
-} from 'me/concierge/constants';
+import { updateConciergeAppointmentDetails } from 'state/concierge/actions';
+import { CONCIERGE_APPOINTMENT_DETAILS_REQUEST } from 'state/action-types';
+import { noRetry } from 'state/data-layer/wpcom-http/pipeline/retry-on-failure/policies';
 import fromApi from './from-api';
+import bookHandlers from './book';
+import cancelHandlers from './cancel';
+import rescheduleHandlers from './reschedule';
 
-export const toApi = ( { beginTimestamp, customerId, siteId, meta } ) => ( {
-	begin_timestamp: beginTimestamp / 1000, // convert to UNIX timestamp.
-	customer_id: customerId,
-	site_id: siteId,
-	meta: JSON.stringify( meta ),
-} );
-
-export const cancelConciergeAppointment = ( { dispatch }, action ) => {
-	dispatch( updateConciergeBookingStatus( CONCIERGE_STATUS_CANCELLING ) );
+export const fetchAppointmentDetails = ( { dispatch }, action ) => {
+	const { appointmentId, scheduleId } = action;
 
 	dispatch(
 		http(
 			{
-				method: 'POST',
-				path: `/concierge/schedules/${ action.scheduleId }/appointments/${
-					action.appointmentId
-				}/cancel`,
+				method: 'GET',
+				path: `/concierge/schedules/${ scheduleId }/appointments/${ appointmentId }/detail`,
 				apiNamespace: 'wpcom/v2',
-				body: {},
+				retryPolicy: noRetry(),
```

Comment:
Unless a very special case to not retry, I think we can rely on the default retry police here. See https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/wpcom-http/pipeline/retry-on-failure/README.md#usage.

Reply:
I added the noRetry strategy because the endpoint returns 404 if the appointment is not found which will trigger 3 automatic retries. We don't really need those because we show an error and recommend a full refresh.

Reply:
Hmm, interesting. I personally think this is not the right behavior, since 404 means "yes, your request is correct but we just don't have this thing for you.", so there should be no value firing retry for it. Maybe we should fix that in another PR.

cc @dmsnell here in case I'm just insane :)